src/application/components/Enum.js

```diff
@@ -0,0 +1,41 @@
+// @flow
```

Comment:
Hmm duplication, i forget its not golang sometimes :D I do think it would make Model.js more too complex and I think you can display enums differently to models potentially. Right now I feel it fits with the flow type, one takes a model type and the other take an enum type. Theres some slightly different logic between the two in how it renders the fields.